diff --git a/Source/JavaScriptCore/assembler/TargetAssemblerDefinitions.h b/Source/JavaScriptCore/assembler/TargetAssemblerDefinitions.h
index 27d11b1e23..5f28546cf8 100644
--- a/Source/JavaScriptCore/assembler/TargetAssemblerDefinitions.h
+++ b/Source/JavaScriptCore/assembler/TargetAssemblerDefinitions.h
@@ -82,6 +82,9 @@ WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
 #elif CPU(RISCV64)
 #define TARGET_ASSEMBLER RISCV64Assembler
 
+#elif CPU(PPC64LE)
+#define TARGET_ASSEMBLER PPC64Assembler
+
 #else
 #error "The MacroAssembler is not supported on this platform."
 #endif
