diff --git a/Source/JavaScriptCore/domjit/DOMJITEffect.h b/Source/JavaScriptCore/domjit/DOMJITEffect.h
index 6ba85116d1..94bbd07251 100644
--- a/Source/JavaScriptCore/domjit/DOMJITEffect.h
+++ b/Source/JavaScriptCore/domjit/DOMJITEffect.h
@@ -44,6 +44,7 @@ struct Effect {
         return { readRange, HeapRange::none() };
     }
 
+#if ENABLE(DFG_JIT)
     constexpr static Effect forReadKinds(DFG::AbstractHeapKind read1 = DFG::InvalidAbstractHeap, DFG::AbstractHeapKind read2 = DFG::InvalidAbstractHeap, DFG::AbstractHeapKind read3 = DFG::InvalidAbstractHeap, DFG::AbstractHeapKind read4 = DFG::InvalidAbstractHeap)
     {
         return { HeapRange::none(), HeapRange::none(), HeapRange::none(), { read1, read2, read3, read4 } };
@@ -59,6 +60,8 @@ struct Effect {
         return { HeapRange::none(), HeapRange::none(), HeapRange::none(), { reads[0], reads[1], reads[2], reads[3] }, { writes[0], writes[1], writes[2], writes[3] } };
     }
 
+#endif
+
     constexpr static Effect forReadWrite(HeapRange readRange, HeapRange writeRange)
     {
         return { readRange, writeRange };
@@ -86,18 +89,23 @@ struct Effect {
 
     constexpr bool isTop() const
     {
-        return domWrites == HeapRange::top() ||
-            writes[0] == DFG::Heap ||
-            writes[1] == DFG::Heap ||
-            writes[2] == DFG::Heap ||
-            writes[3] == DFG::Heap;
+        return domWrites == HeapRange::top()
+#if ENABLE(DFG_JIT)
+            || writes[0] == DFG::Heap
+            || writes[1] == DFG::Heap
+            || writes[2] == DFG::Heap
+            || writes[3] == DFG::Heap
+#endif
+            ;
     }
 
     HeapRange domReads { HeapRange::top() };
     HeapRange domWrites { HeapRange::top() };
     HeapRange def { HeapRange::top() };
+#if ENABLE(DFG_JIT)
     DFG::AbstractHeapKind reads[4] { DFG::InvalidAbstractHeap };
     DFG::AbstractHeapKind writes[4] { DFG::InvalidAbstractHeap };
+#endif
 };
 
 }
