diff --git a/Source/JavaScriptCore/jit/JITArithmetic.cpp b/Source/JavaScriptCore/jit/JITArithmetic.cpp
index a7ddc8346f..440b7e47e4 100644
--- a/Source/JavaScriptCore/jit/JITArithmetic.cpp
+++ b/Source/JavaScriptCore/jit/JITArithmetic.cpp
@@ -518,7 +518,7 @@ void JIT::emitSlow_op_mod(const JSInstruction*, Vector<SlowCaseEntry>::iterator&
     slowPathCall.call();
 }
 
-#elif CPU(ARM64)
+#elif CPU(ARM64) || CPU(PPC64LE)
 
 void JIT::emit_op_mod(const JSInstruction* currentInstruction)
 {
@@ -567,7 +567,7 @@ void JIT::emitSlow_op_mod(const JSInstruction*, Vector<SlowCaseEntry>::iterator&
     slowPathCall.call();
 }
 
-#else // CPU(X86_64) || CPU(ARM64)
+#else // CPU(X86_64) || CPU(ARM64) || CPU(PPC64LE)
 
 void JIT::emit_op_mod(const JSInstruction*)
 {
@@ -580,7 +580,7 @@ void JIT::emitSlow_op_mod(const JSInstruction*, Vector<SlowCaseEntry>::iterator&
     UNREACHABLE_FOR_PLATFORM();
 }
 
-#endif // CPU(X86_64) || CPU(ARM64)
+#endif // CPU(X86_64) || CPU(ARM64) || CPU(PPC64LE)
 
 /* ------------------------------ END: OP_MOD ------------------------------ */
 
