diff --git a/Source/JavaScriptCore/runtime/Options.h b/Source/JavaScriptCore/runtime/Options.h
index 80a5b11f51..1080b7f681 100644
--- a/Source/JavaScriptCore/runtime/Options.h
+++ b/Source/JavaScriptCore/runtime/Options.h
@@ -175,7 +175,7 @@ private:
     static unsigned computeNumberOfWorkerThreads(int maxNumberOfWorkerThreads, int minimum = 1);
     static int32_t computePriorityDeltaOfWorkerThreads(int32_t twoCorePriorityDelta, int32_t multiCorePriorityDelta);
     static constexpr bool jitEnabledByDefault() { return isAddress64Bit(); }
-    static constexpr bool ipintEnabledByDefault() { return isARM64() || isARM64E() || isX86_64(); }
+    static constexpr bool ipintEnabledByDefault() { return isARM64() || isARM64E() || isX86_64() || isPPC64(); }
     static double defaultQuickDFGTierUpThresholdFactor()
     {
 #if PLATFORM(MAC)
